### PR TITLE
fix example flask-demo and add a simple client

### PR DESCRIPTION
**Pull request by afuechsel**
These changes fix the flask-demo, that contained several syntax errors. I will add further comments in the pull request. It also adds a simple client using the `request` package, that is mentioned in the course but never really shown.

### CODE REVIEW

File: lesson-3-implementing-message-passing/flask-demo/app.py [hunk 1]
```diff
@@ -7,15 +7,15 @@
 
 @app.route('/')
 def hello_world():
-    return 'Hello World!'
+    return jsonify({'message': 'Hello World!'})
```
Comment (afuechsel) [inline lesson-3-implementing-message-passing/flask-demo/app.py:10]: As `jsonify` has been imported originally I thought you want to use it too.

File: lesson-3-implementing-message-passing/flask-demo/app.py [hunk 2]
```diff
@@ -7,15 +7,15 @@
...
 
 @app.route('/demo/<path_demo>', methods=['POST'])
 def demo(path_demo=None):
     # Showcase different ways in which data can be passed
     path_value = path_demo
     header_value = request.headers.get('header_demo', None)
     param_value = request.args.get('param_demo', None)
-    body_value = request.data
+    body_value = request.data.decode('UTF-8')
```
Comment (afuechsel) [inline lesson-3-implementing-message-passing/flask-demo/app.py:18]: Without that python is printing the string with a leading `b'` to indicate, that it's binary. Confusing to new users. See https://stackoverflow.com/questions/57337321/flask-b-text-appears-before-request-data-results.

File: lesson-3-implementing-message-passing/flask-demo/app.py [hunk 3]
```diff
@@ -28,13 +28,12 @@ def demo(path_demo=None):
 @app.route('/api/items/<item_id>', methods=['GET', 'POST'])
 def pets(item_id):
     if request.method == 'GET':
-        return Response(json.dumps(retrieve_item(pet_id)), 200, {'Content-Type': 'application/json'})
+        return Response(json.dumps(retrieve_item(item_id)), 200, {'Content-Type': 'application/json'})
```
Comment (afuechsel) [inline lesson-3-implementing-message-passing/flask-demo/app.py:31]: pet_id was nowehere defined

File: lesson-3-implementing-message-passing/flask-demo/services.py [hunk 4]
```diff
@@ -10,13 +10,11 @@ def retrieve_item(item_id):
     }
 
 
-def create_item(item_id):
+def create_item(item_id, request_body):
     """
     This is a stubbed method of creating a resource. It doesn't actually do anything.
     """
     return {
```
Comment (afuechsel) [inline lesson-3-implementing-message-passing/flask-demo/services.py:17]: It should at least simulate, that you request_body has been processed somehow.

**Comment by SudKul** [comment]
All proposed changes look good. 
Thanks, @afuechsel

**Pull request merged**